src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -149,55 +193,159 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
+            var nativeToManagedVtables =
+                generateStubInformation
+                .Collect()
+                .SelectMany(static (data, ct) => GroupContextsForInterfaceGeneration(data.CastArray<GeneratedMethodContextBase>()))
+                .Select(static (vtable, ct) => GenerateImplementationVTable(ImmutableArray.CreateRange(vtable.Array.Cast<IncrementalMethodStubGenerationContext>())))
+                .WithTrackingName(StepNames.GenerateNativeToManagedVTable)
+                .WithComparer(SyntaxEquivalentComparer.Instance)
+                .SelectNormalized();
+
+            var iUnknownDerivedAttributeApplication = interfaceWithMethodsContexts
+                .Select(static (context, ct) => GenerateIUnknownDerivedAttributeApplication(context))
+                .WithTrackingName(StepNames.GenerateIUnknownDerivedAttribute)
+                .WithComparer(SyntaxEquivalentComparer.Instance)
+                .SelectNormalized();
+
+            var filesToGenerate = interfaceWithMethodsContexts
+                .Zip(nativeInterfaceInformation)
+                .Zip(managedToNativeInterfaceImplementations)
+                .Zip(nativeToManagedVtableMethods)
+                .Zip(nativeToManagedVtables)
+                .Zip(iUnknownDerivedAttributeApplication)
+                .Select(static (data, ct) =>
+                {
+                    var (((((interfaceContext, interfaceInfo), managedToNativeStubs), nativeToManagedStubs), nativeToManagedVtable), iUnknownDerivedAttribute) = data;
+
+                    using StringWriter source = new();
+                    interfaceInfo.WriteTo(source);
+                    // Two newlines looks cleaner than one
+                    source.WriteLine();
+                    source.WriteLine();
+                    // TODO: Merge the three InterfaceImplementation partials? We have them all right here.
+                    managedToNativeStubs.WriteTo(source);
+                    source.WriteLine();
+                    source.WriteLine();
+                    nativeToManagedStubs.WriteTo(source);
+                    source.WriteLine();
+                    source.WriteLine();
+                    nativeToManagedVtable.WriteTo(source);
+                    source.WriteLine();
+                    source.WriteLine();
+                    iUnknownDerivedAttribute.WriteTo(source);
+                    return new { TypeName = interfaceContext.InterfaceType.FullTypeName, Source = source.ToString() };
+                });
+
+            context.RegisterSourceOutput(filesToGenerate, (context, data) =>
+            {
+                context.AddSource(data.TypeName.Replace("global::", ""), data.Source);
+            });
+        }
 
-            context.RegisterConcatenatedSyntaxOutputs(generateNativeInterface, "NativeInterfaces.g.cs");
+        private static string GenerateMarkerInterfaceSource(ComInterfaceContext iface) => $$"""
```

Comment:
Is this something we'll want to switch to using the SyntaxFactory methods, or should it stay as an interpolated string?

Reply:
I think this is a lot cleaner as an interpolated string and making it typed with SyntaxFactory doesn't get us the benefits of syntax nodes that we get in the rest of the project (more structure around wrapping syntax and knowing what type of syntax we're working with at various API boundaries). I'd be curious what @AaronRobinsonMSFT and @elinor-fung think for this case.